### PR TITLE
restore CI output group folding in github actions

### PR DESCRIPTION
**Pull request by SubaruArai**
Related: #25334

See official docs for details:
https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#grouping-log-lines

### CODE REVIEW

**Review by audrow — APPROVED** [review]
This looks like a good change to me and makes the CI easier to go through. @clalancette, what do you think?

File: test/fold_block.py
Comment (clalancette) [inline test/fold_block.py:49]: This bit can be removed; we are not going to go back to `travis`.

**Review by clalancette — CHANGES_REQUESTED** [review]
I've left one thing to change in line, but this looks great.

File: test/fold_block.py
Comment (SubaruArai) [reply test/fold_block.py:49]: I first thought to delete it, but it doesn't hurt being there, so...
@audrow what do you think?

Comment (audrow) [reply test/fold_block.py:49]: If we're not using it, I'd favor removing it.

The reason is that I'd rather not have unused code hanging around.

**Comment by SubaruArai** [comment]
Sorry, I just added an update.
I didn't like that `cls_` remained in the module.

**Review by clalancette — APPROVED** [review]
This seems good to me, but I'll wait for another review from @audrow before merging in.

**Review by audrow — APPROVED** [review]
Looks good to me!

**Pull request merged**
